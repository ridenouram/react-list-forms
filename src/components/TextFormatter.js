import React, { PureComponent } from 'react';

export default class TextFormatter extends PureComponent {
  state = {
    text: '',
    color: 'black'
  };

  //change handlers
  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  render() {
    const { text, color } = this.state;

    return (
      <form>
        <input type="text" name="text" value={text} onChange={this.handleChange} />
        <input type="color" name="color" value={color} onChange={this.handleChange} />
        <p style={{ color: color }}>{text}</p>
      </form>
    );
  }
}
