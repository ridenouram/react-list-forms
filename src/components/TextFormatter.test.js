import React from 'react';
import { shallow } from 'enzyme';
import TextFormatter from './TextFormatter';

describe('TextFormatter component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<TextFormatter />);
    expect(wrapper).toMatchSnapshot();
  });
});
