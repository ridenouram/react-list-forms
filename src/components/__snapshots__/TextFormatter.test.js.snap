// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFormatter component matches snapshot 1`] = `
<form>
  <input
    name="text"
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    name="color"
    onChange={[Function]}
    type="color"
    value="black"
  />
  <p
    style={
      Object {
        "color": "black",
      }
    }
  />
</form>
`;
